### PR TITLE
[DX-936] Add new-command skill and adopt output helpers across CLI

### PR DESCRIPTION
**Pull request by umair-ably**
## Summary

- **New-command skill**: Adds `.claude/skills/ably-new-command/` with `SKILL.md` and reference docs (`patterns.md`, `testing.md`) that guide Claude through scaffolding new CLI commands with correct base classes, flag sets, output patterns, error handling, and test scaffolds.
- **Output helper refactoring**: Prefixed all output helpers with `format` (e.g., `formatSuccess`, `formatProgress`, `formatLabel`) for consistency and to reduce import clashes. Adopted `formatLabel()` across 11 command files that were using raw label strings or `chalk.dim("Label:")`.
- **Code fixes**: Removed the `formatSuccess as successMsg` alias in `connections/test.ts`. Replaced `chalk.green("Integration Rule Details:")` with `formatHeading()` in `integrations/get.ts`.
- **New tests**: Added `test/unit/utils/output.test.ts` with 35 contract tests covering all format helpers (append behavior, pluralization, color wrapping, edge cases).
- **Skill doc improvements**: Shortened frontmatter, consolidated error handling guidance around `handleCommandError`, added Realtime publish guidance, fixed `describe.skipIf` pattern in E2E test scaffolds, documented `setupChannelStateLogging` helper.

### Skill documentation
- `.claude/skills/ably-new-command/SKILL.md` — shortened frontmatter, consolidated error handling, clarified auth guidance
- `.claude/skills/ably-new-command/references/patterns.md` — channel state helper docs, Realtime publish guidance
- `.claude/skills/ably-new-command/references/testing.md` — fixed `describe.skipIf` E2E pattern

### Source code (`formatLabel` adoption)
- `src/commands/apps/create.ts`
- `src/commands/apps/delete.ts`
- `src/commands/apps/update.ts`
- `src/commands/apps/set-apns-p12.ts`
- `src/commands/auth/keys/create.ts`
- `src/commands/queues/create.ts`
- `src/commands/queues/delete.ts`
- `src/commands/integrations/create.ts`
- `src/commands/integrations/delete.ts`
- `src/commands/integrations/get.ts`
- `src/commands/spaces/locks/get-all.ts`

### Source code (other fixes)
- `src/commands/connections/test.ts` — removed `successMsg` alias

### New tests
- `test/unit/utils/output.test.ts` — 35 contract tests for format helpers

## Test plan

- [x] `pnpm prepare` succeeds
- [x] `pnpm exec eslint .` — 0 errors
- [x] `pnpm test:unit` — 1150 tests pass (123 test files)
- [ ] Verify formatted labels render correctly: `node bin/run.js apps create --name "test-app" --json` (check non-JSON output has dim labels)
- [ ] Spot-check a few commands for correct label formatting in terminal

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/skills/ably-new-command/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+}
+```
+
+Rules:
+- `progress("Action text")` — appends `...` automatically, never add it manually
+- `success("Completed action.")` — green checkmark, always end with `.` (period, not `!`)
+- `listening("Listening for X.")` — dim text, automatically appends "Press Ctrl+C to exit."
+- `resource(name)` — cyan colored, never use quotes around resource names
+- `formatTimestamp(ts)` — dim `[timestamp]` for event streams
+- `chalk.blue(clientId)` for client IDs
```
Comment (AndyTWF) [inline .claude/skills/ably-new-command/SKILL.md]: For arguments sake, should we give these "names" rather than just chalk.blue?

File: .claude/skills/ably-new-command/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+  }
+}
+```
+
+**Structured record lists** (e.g., `queues list`, `integrations list`, `push devices list`) — use `chalk.bold()` as a record heading with detail fields below:
+```typescript
+if (!this.shouldOutputJson(flags)) {
+  this.log(`Found ${items.length} devices:\n`);
+  for (const item of items) {
+    this.log(chalk.bold(`Device ID: ${item.id}`));
```
Comment (AndyTWF) [inline .claude/skills/ably-new-command/SKILL.md]: Again separate PR probably but having named functions for heading, listItem etc might be worth it?
